drv/ArrayFIFOQueue.drv

```diff
@@ -56,6 +56,23 @@ public class ARRAY_FIFO_QUEUE KEY_GENERIC implements PRIORITY_QUEUE KEY_GENERIC,
 	 *  Might be actually smaller than {@link #start} because {@link #array} is used cyclically. */
 	protected transient int end;
 
+	public ARRAY_FIFO_QUEUE(final KEY_GENERIC_TYPE[] values) {
+		this(values, 0, values.length);
+	}
+	
+	public ARRAY_FIFO_QUEUE(final KEY_GENERIC_TYPE[] values, int size) {
+		this(values, 0, size);
+	}
+
+	public ARRAY_FIFO_QUEUE(final KEY_GENERIC_TYPE[] values, int offset, int size) {
+		if (values.length < 0) throw new IllegalArgumentException("Initial array (" + values.length + ") is negative");
```

Comment:
arrays can't have negative size. Rather check if size < 0 or size > values.length I guess

Reply:
Yeah, I can see and agree why that is a better idea.